nsq/async.py

```diff
@@ -277,7 +277,7 @@ def _read_body(self, data):
             self.trigger(event.DATA, conn=self, data=data)
         except Exception:
             logger.exception('uncaught exception in data event')
-        self.io_loop.add_callback(self._start_read)
```

Comment:
Right -- the question I have is whether it's possible that enough data could be fed in at the right rhythm that would cause it to never defer. That's something I'm a bit unsure about.